scribus/ui/fontcombo.cpp

```diff
@@ -320,3 +322,101 @@ void FontComboH::RebuildList(ScribusDoc *currentDoc, bool forAnnotation, bool fo
 	fontFamily->blockSignals(familySigBlocked);
 	fontStyle->blockSignals(styleSigBlocked);
 }
+
+fontFamilyDelegate::fontFamilyDelegate(QObject *parent):
+	QAbstractItemDelegate(parent)
+{
+}
+
+void fontFamilyDelegate::paint(QPainter *painter,
+							   const QStyleOptionViewItem &option,
+							   const QModelIndex &index) const
+{
+	QString text = index.data(Qt::DisplayRole).toString();
+	QFont font;
+	QStringList families = QFontDatabase().families();
+	QString name, style;
+	for (int i =0; i < families.size(); ++i)
```

Comment:
The idea here, I am depending on QFontdatabase to get the right family name. The logic here:
1- Get all font families from QFontdatabase.
2- Exam the whole text against all families name.
3- if match then assign the font.
In you case, let assume that we have two fonts like this: `Foo` & `Foo Bar`
the text will be something like this: `Foo regular ` and ` Foo Bar regular`
Here is how it works for `Foo regular`:
family name from Qfontdatabase is `Foo`
then, if `Foo regular` starts with `Foo` then font is `Foo`. 
the second loop is false because `Foo regular` doesn't start with `Foo Bar`
Let see how works with `Foo Bar regular`:
family name from Qfontdatabase is `Foo Bar`
then,  `Foo Bar  regular` starts with `Foo` then font is `Foo` same like above
the second loop is true because `Foo Bar regular`  start with `Foo Bar` and since QfonDatabase store fonts in order, font will be `Foo Bar` which is right.

I know it is tricky but without QfontDatabase things will be very complicated. If you have another idea please share it.